src/SLCore.UnitTests/Analysis/SLCoreAnalyzerTests.cs

```diff
@@ -0,0 +1,233 @@
+        
+        testSubject.ExecuteAnalysis(@"C:\file\path", Guid.NewGuid(), default, default, default, default, default);
+
+        analysisStatusNotifierFactory.Received().Create(nameof(SLCoreAnalyzer), @"C:\file\path");
+        notifier.Received().AnalysisStarted();
+    }
+    
+    [TestMethod]
+    public void ExecuteAnalysis_ConfigScopeNotInitialized_NotifyNotReady()
+    {
+        var activeConfigScopeTracker = Substitute.For<IActiveConfigScopeTracker>();
+        activeConfigScopeTracker.Current.Returns((ConfigurationScope)null);
+        var testSubject = CreateTestSubject(CreatServiceProvider(out var analysisService), activeConfigScopeTracker, CreateDefaultAnalysisStatusNotifier(out var notifier));
+        
+        testSubject.ExecuteAnalysis(@"C:\file\path", Guid.NewGuid(), default, default, default, default, default);
+
+        _ = activeConfigScopeTracker.Received().Current;
+        analysisService.ReceivedCalls().Should().BeEmpty();
+        notifier.Received().AnalysisNotReady(SLCoreStrings.ConfigScopeNotInitialized);
+    }
+    
+    [TestMethod]
+    public void ExecuteAnalysis_ServiceProviderUnavailable_NotifyFailed()
+    {
+        var slCoreServiceProvider = CreatServiceProvider(out var analysisService, false);
+        var testSubject = CreateTestSubject(slCoreServiceProvider, CreateInitializedConfigScope("someconfigscopeid"), CreateDefaultAnalysisStatusNotifier(out var notifier));
+        
+        testSubject.ExecuteAnalysis(@"C:\file\path", Guid.NewGuid(), default, default, default, default, default);
+
+        slCoreServiceProvider.Received().TryGetTransientService(out Arg.Any<IAnalysisSLCoreService>());
+        analysisService.ReceivedCalls().Should().BeEmpty();
+        notifier.Received().AnalysisFailed(SLCoreStrings.ServiceProviderNotInitialized);
+    }
+    
+    [TestMethod]
+    public void ExecuteAnalysis_PassesCorrectArgumentsToAnalysisService()
+    {
+        var timestampTestStart = DateTimeOffset.Now.ToUnixTimeMilliseconds();
+        var analysisId = Guid.NewGuid();
+        var testSubject = CreateTestSubject(CreatServiceProvider(out var analysisService), CreateInitializedConfigScope("someconfigscopeid"));
+
+        testSubject.ExecuteAnalysis(@"C:\file\path", analysisId, default, default, default, default, default);
+
+        var timestampTestAssert = DateTimeOffset.Now.ToUnixTimeMilliseconds();
+        analysisService.Received().AnalyzeFilesAndTrackAsync(Arg.Is<AnalyzeFilesAndTrackParams>(a => 
+                a.analysisId == analysisId 
+                && a.configurationScopeId == "someconfigscopeid"
+                && a.filesToAnalyze.Single() == new FileUri(@"C:\file\path")
+                && a.extraProperties != null
+                && a.shouldFetchServerIssues
+                && a.startTime >= timestampTestStart
+                && a.startTime <= timestampTestAssert),
+            Arg.Any<CancellationToken>());
```

Comment:
Pass a cancellation token to ExecuteAnalysis and check if it is propagated here.

Reply:
The test is below